src/librustc_resolve/resolve_imports.rs

```diff
@@ -587,7 +587,21 @@ impl<'a, 'b:'a> ImportResolver<'a, 'b> {
         let module = match module_result {
             Success(module) => module,
             Indeterminate => return Indeterminate,
-            Failed(err) => return Failed(err),
+            Failed(err) => {
+                let self_module = self.current_module.clone();
```

Comment:
This could just be `let self_module = self.current_module`.

However, https://github.com/rust-lang/rust/issues/34191#issuecomment-241935757 isn't entirely correct (sorry). If we are in a block that has items in it, `self.current_module` is the anonymous module associated with the block, but `self` is the "normal" (`mod`) module that contains the block.

To get the `self` module, you can use `self.module_map[&self.current_module.normal_ancestor_id.unwrap()]`
(c.f. `fn resolve_module_prefix` in `librustc_resolve/lib.rs`).